src/transports/ws/sws.c

```diff
@@ -70,9 +71,9 @@
 #define NN_WS_OPCODE_UNUSED5 0x05
 #define NN_WS_OPCODE_UNUSED6 0x06
 #define NN_WS_OPCODE_UNUSED7 0x07
-#define NN_WS_OPCODE_CLOSE NN_WS_MSG_TYPE_CLOSE
-#define NN_WS_OPCODE_PING NN_WS_MSG_TYPE_PING
-#define NN_WS_OPCODE_PONG NN_WS_MSG_TYPE_PONG
+#define NN_WS_OPCODE_CLOSE 0x08
+#define NN_WS_OPCODE_PING 0x09
+#define NN_WS_OPCODE_PONG 0x0A
```

Comment:
Expanding; even though they are part of the official RFC 6455 spec, and even though they currently exist within libnanomsg, they should probably be encapsulated within the transport layer, rather than exposed via the application layer.

Basically, their existence, like the Close Handshake, and also message types of either Binary or UTF-8, really throw a wrench into things for this library :-\